api/v1alpha1/driver_types.go

```diff
@@ -0,0 +1,268 @@
+	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy"`
+}
+
+type ProvisionerResourcesSpec struct {
+	Attacher      *corev1.ResourceRequirements `json:"attacher,omitempty"`
+	Snapshotter   *corev1.ResourceRequirements `json:"snapshotter,omitempty"`
+	Resizer       *corev1.ResourceRequirements `json:"resizer,omitempty"`
+	Provisioner   *corev1.ResourceRequirements `json:"provisioner,omitempty"`
+	OMapGenerator *corev1.ResourceRequirements `json:"omapGenerator,omitempty"`
+	Liveness      *corev1.ResourceRequirements `json:"liveness,omitempty"`
+	Plugin        *corev1.ResourceRequirements `json:"plugin,omitempty"`
+}
+
+type ProvisionerSpec struct {
+	// Embedded common pods spec
+	PodCommonSpec `json:"inline"`
+
+	// Set replicas for csi provisioner deployment. Defaults to 2
+	Replicas *int32 `json:"provisionerReplicas,omitempty"`
+
+	// Resource requirements for provisioner's containers
+	Resources ProvisionerResourcesSpec `json:"resources,omitempty"`
+}
+
+type LivenessSpec struct {
+	// Port to expose liveness metrics
+	MetricsPort int `json:"metricsPort,omitempty"`
+}
+
+type LeaderElectionSpec struct {
+	// Duration in seconds that non-leader candidates will wait to force acquire leadership.
+	// Default to 137 seconds.
+	LeaseDuration int `json:"leaseDuration,omitempty"`
+
+	// Deadline in seconds that the acting leader will retry refreshing leadership before giving up.
+	// Defaults to 107 seconds.
+	RenewDeadline int `json:"renewDeadline,omitempty"`
+
+	// Retry Period in seconds the LeaderElector clients should wait between tries of actions.
+	// Defaults to 26 seconds.
+	RetryPeriod int `json:"retryPeriod,omitempty"`
+}
+
+type CephFsClientType string
+
+const (
+	KernelCephFsClient CephFsClientType = "kernel"
+	FuseCephFsClient   CephFsClientType = "fuse"
+)
+
+// DriverSpec defines the desired state of Driver
+type DriverSpec struct {
+	// Logging configuration for driver's pods
+	Logging *LogSpec `json:"logging,omitempty"`
+
+	// A reference to a ConfigMap resource holding image overwrite for deployed
+	// containers
+	ImageSet *corev1.LocalObjectReference `json:"imageSet,omitempty"`
+
+	// Cluster name identifier to set as metadata on the CephFS subvolume and RBD images. This will be useful in cases
+	// when two container orchestrator clusters (Kubernetes/OCP) are using a single ceph cluster.
+	ClusterName *string `json:"clusterName,omitempty"`
+
+	// Set to true to enable adding volume metadata on the CephFS subvolumes and RBD images.
+	// Not all users might be interested in getting volume/snapshot details as metadata on CephFS subvolume and RBD images.
+	// Hence enable metadata is false by default.
+	EnableMetadata *bool `json:"enableMetadata,omitempty"`
+
+	// Set the gRPC timeout for gRPC call issued by the driver components
+	GRpcTimeout int `json:"grpcTimeout,omitempty"`
+
+	// Select a policy for sanpshot behavior: none, autodetect, snapshot, sanpshotGroup
```

Comment:
`sanpshotGroup`  to `snapshotGroup` we can also go for `volumeSnapshot` and volumeGroupSnapshot` which are the terms used in kubernetes.

Reply:
Ok that is batter, I will change to `volumeSnapshot` and `volumeGroupSnapshot`

Reply:
done